src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,438 @@
+                   "CubeMap::saveTexture(): Unknown texture type.", false);
+
+    if (!converter->exportToFile(image, filename)) {
+      return false;
+    } else {
+      LOG(INFO) << "Saved image " << iFace << " to " << filename;
+    }
+  }
+
+  return true;
+}
+#endif
+
+void CubeMap::loadTexture(TextureType type,
+                          const std::string& imageFilePrefix,
+                          const std::string& imageFileExtension) {
+  textureTypeSanityCheck(flags_, type, "CubeMap::loadTexture():");
+
+  // plugin manager used to instantiate importers which in turn are used
+  // to load image data
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> manager;
+  Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+      manager.loadAndInstantiate("AnyImageImporter");
+  CORRADE_INTERNAL_ASSERT(importer);
+
+  const char* coordStrings[6] = {"+X", "-X", "+Y", "-Y", "+Z", "-Z"};
+  int imageSize = 0;
+
+  // set images
+  Mn::GL::CubeMapTexture* texture = nullptr;
+  switch (type) {
+    case TextureType::Color:
+      texture = textures_[TextureType::Color].get();
+      break;
+
+    case TextureType::Depth:
+      texture = textures_[TextureType::Depth].get();
+      break;
+  }
+  CORRADE_ASSERT(texture, "CubeMap::loadTexture(): Unknown texture type.", );
+
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    // open image file
+
+    std::string filename = Cr::Utility::formatString(
+        "{}.{}.{}.{}", imageFilePrefix, getTextureTypeFilenameString(type),
+        coordStrings[iFace], imageFileExtension);
+
+    importer->openFile(filename);
+    Cr::Containers::Optional<Mn::Trade::ImageData2D> imageData =
+        importer->image2D(0);
+
+    // sanity checks
+    CORRADE_INTERNAL_ASSERT(imageData);
+    Mn::Vector2i size = imageData->size();
+    CORRADE_ASSERT(
+        size.x() == size.y(),
+        " CubeMap::loadTexture(): each texture image must be a square.", );
+    if (iFace == 0) {
+      imageSize = size.x();
+      reset(imageSize);
+    } else {
+      CORRADE_ASSERT(size.x() == imageSize,
+                     " CubeMap::loadTexture(): texture images must have the "
+                     "same size.", );
+    }
+
+    switch (type) {
+      case TextureType::Color:
+        texture->setSubImage(convertFaceIndexToCubeMapCoordinate(iFace), 0, {},
+                             *imageData);
+        break;
+
+      case TextureType::Depth: {
+        // The pixel format for depth texture is R32F. When it is saved as hdr,
+        // the single channel is expanded to three channels by repeating the R
+        // channel 3 times (and it becomes RGB32F).
+        // That means when it is loaded, we need to dedup by taking just the
+        // first component (R component) out of each pixel.
+        Cr::Containers::StridedArrayView2D<const Mn::Color3>
+            imageStridedArrayView = imageData->pixels<Mn::Color3>();
+
+        Cr::Containers::Array<float> depthImage{
```

Comment:
```suggestion
        Cr::Containers::Array<float> depthImage{Cr::Containers::NoInit,
```

to avoid zero-initializing the whole array, since it's overwritten anyway -- for bigger sizes the time saving is quite significant